File: tests/Fortran/Free/0011.f90
program dot
  implicit none
  integer, parameter :: N = 1024
  integer i
  real :: x(N), y(N)
  real :: dot1, dot2

  !$acc parallel loop reduction(+:dot1)
  do i = 1, N
    dot1 = x(i)*y(i)
  end do

  !$acc parallel loop reduction(+:dot1) reduction(+:dot2)
  do i = 1, N
    dot1 = x(i)*y(i)
    dot2 = y(i)*x(i)
  end do

end program

File: tests/C/0011.c
float dot (int N, float * x, float * y)
{
	float tmp = 0;
	#pragma acc parallel loop reduction(+:tmp)
	for (int i = 0; i < N; ++i)
		tmp += x[i] * y[i];
	return tmp;
}

void dot2 (int N, float * x, float * y, float *d1, float *d2)
{
	float tmp1 = 0, tmp2 = 0;
	#pragma acc parallel loop reduction(+:tmp1) reduction(+:tmp2)
	for (int i = 0; i < N; ++i)
	{
		tmp1 += x[i] * y[i];
		tmp2 += y[i] * x[i];
	}
	*d1 = tmp1;
	*d2 = tmp2;
}
